onnxruntime/python/tools/kernel_explorer/kernels/gemm.cc

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <type_traits>
+
+#include <pybind11/pybind11.h>
+
+#include "core/providers/rocm/rocm_common.h"
+#include "device_array.h"
+#include "operator.h"
+
+#include "core/providers/rocm/shared_inc/fpgeneric.h"
+
+namespace py = pybind11;
+
+namespace onnxruntime {
+
+enum class BlasOp {
+  N,
+  T,
+  // C,
+};
+
+// assume row majored
```

Comment:
Where is this assumption reflected in the code? I think cuBLAS/rocBLAS assumes inputs are column-major? (https://stackoverflow.com/questions/56043539/cublassgemm-row-major-multiplication)

Reply:
Yeah, both cuBLAS and rocBLAS assume column-majored matrix. Only oneMKL supports both layouts.

Reply:
I'm a bit confused. Should the comment be "assume column-majored" then?

Reply:
Yeah, it is concusing. I think I should add more comment to clarify the rocblas/cublas is an exception. Or we adapt the interface to unify it.

Since in deeplearning, generally speaking tensors are row-majored (last dim consecutive). Calling into BLAS subroutine with column-majored convention, it is only a matter of swapping position of the A and the B matrix and related params. We can make all pybind interfaces have the same tensor semantic if we adapt it before we calling into the BLAS subroutine.

Reply:
Done